umake/frameworks/ide.py

```diff
@@ -755,3 +755,84 @@ def post_install(self):
+    def get_metadata(self, result):
+        """Download files to download"""
+        logger.debug("Parse download metadata")
+
+        error_msg = result[self.download_page].error
+        if error_msg:
+            logger.error("An error occurred while downloading {}: {}".format(self.download_page, error_msg))
+            UI.return_main_screen(status_code=1)
+
+        in_download = False
+        url_found = False
+        line_content = result[self.download_page].buffer.getvalue().decode()
+        (url_found, in_download) = self.parse_download_link(line_content, in_download)
+
+        if not url_found:
+            logger.error("Download page changed its syntax or is not parsable")
+            UI.return_main_screen(status_code=1)
+
+    def parse_download_link(self, line, in_download):
+        """Parse STS download link"""
+        url_found = False
+        in_download = False
+
+        if '>Linux<' in line and '64BIT' in line:
+            in_download = True
+        if in_download:
+            regexp = r'GTK, 64BIT.+?href="(.*?)"'
+            if get_current_arch() == "i386":
+                regexp = r'GTK, 32BIT.+?href="(.*?)"'
+            r = re.compile(regexp, re.DOTALL)
+            p = r.search(line)
+            with suppress(AttributeError):
+                url_found = True
+                self.sha1_url = '{}.sha1'.format(p.group(1))
+                DownloadCenter(urls=[DownloadItem(self.sha1_url, None)],
+                               on_done=self.get_sha_and_start_download, download=False)
+        return (url_found, in_download)
+
+    @MainLoop.in_mainloop_thread
+    def get_sha_and_start_download(self, download_result):
+        res = download_result[self.sha1_url]
+        sha1 = res.buffer.getvalue().decode('utf-8').split()[0]
+        url = re.sub('.sha1', '', self.sha1_url)
+        if url is None:
+            logger.error("Download page changed its syntax or is not parsable (missing url)")
+            UI.return_main_screen(status_code=1)
+        if sha1 is None:
+            logger.error("Download page changed its syntax or is not parsable (missing sha512)")
+            UI.return_main_screen(status_code=1)
+        logger.debug("Found download link for {}, checksum: {}".format(url, sha1))
+        self.download_requests.append(DownloadItem(url, Checksum(ChecksumType.sha1, sha1)))
+        self.start_download_and_install()
+
+    def post_install(self):
+        """Create the Spring Tools Suite launcher"""
+        categories = "Development;IDE;"
+        create_launcher(self.desktop_filename, get_application_desktop_file(name=_(self.name),
+                                                                            icon_path=os.path.join(self.install_path,
+                                                                                                   self.icon_filename),
+                                                                            exec='"{}" %f'.format(
```

Comment:
could you define the exec_path outside of the create_launcher? We're trying to add links to the executables(to then add to PATH), and we use exec_path.

self.exec_path = os.path.join(self.install_path, "STS")
create_launcher(...
         exec = '"{}" %f'.format(self.exec_path)
       ...